fedimint-core/src/core.rs

```diff
@@ -322,15 +343,53 @@ impl Decoder {
         DecoderBuilder::default()
     }
 
+    /// System Dyn-type, don't use.
+    #[doc(hidden)]
+    pub fn builder_system() -> DecoderBuilder {
+        DecoderBuilder {
+            transparent: true,
+            ..DecoderBuilder::default()
+        }
+    }
+
     /// Decodes a specific `DynType` from the `reader` byte stream.
     ///
     /// # Panics
     /// * If no decoder is registered for the `DynType`
-    pub fn decode<DynType: Any>(
+    pub fn decode_complete<DynType: Any>(
+        &self,
+        reader: &mut dyn Read,
+        total_len: u64,
+        module_id: ModuleInstanceId,
+        decoders: &ModuleDecoderRegistry,
+    ) -> Result<DynType, DecodeError> {
+        let mut reader = reader.take(total_len);
+
+        let val = self.decode_partial(&mut reader, module_id, decoders)?;
+        let left = reader.limit();
```

Comment:
Originally with @maan2003 when debugging, we were printing the bytes here as well, but I decided against it as it opens the code to problems: e.g. what if reading leftover bytes leads to IO errors? Which error we return, which ignore. Also - what if the reader blocks, etc. Seems not worth it.

Reply:
just first print error without bytes, then read bytes and print bytes in seperate message. if all this work, return the error to caller

Reply:
I still don't like it. :D